src/libs/ReportListUtils.js

```diff
@@ -30,63 +47,149 @@ function getParticipantLogins(report) {
+    });
+
+    if (report) {
+        searchTerms.push(...report.reportName.split(',').map(name => name.trim()));
+        searchTerms.push(...report.participants);
+    }
+
+    return _.unique(searchTerms).join(' ');
 }
 
 /**
  * Creates a report list option
  *
  * @param {Array<Object>} personalDetailList
  * @param {Object} [report]
- * @param {Object} [comments]
- * @param {Number} [activeReportID]
  * @returns {Object}
  */
-function createOption(personalDetailList, report, comments, activeReportID) {
-    // There is only one participant so return the single participant view
+function createOption(personalDetailList, report) {
+    const hasMultipleParticipants = personalDetailList.length > 1;
     const personalDetail = personalDetailList[0];
-    const isSingleUserDM = personalDetailList.length < 2;
     return {
         text: report ? report.reportName : personalDetail.displayName,
-        alternateText: (!report || isSingleUserDM) ? personalDetail.login : report.reportName,
+        alternateText: (!report || !hasMultipleParticipants) ? personalDetail.login : report.reportName,
         icons: report ? report.icons : [personalDetail.avatarURL],
-        login: personalDetailList.length < 2 ? personalDetail.login : null,
+
+        // It doesn't make sense to provide a login in the case of a report with multiple participants since
+        // there isn't any one single login to refer to for a report.
+        login: !hasMultipleParticipants ? personalDetail.login : null,
         reportID: report ? report.reportID : null,
         isUnread: report ? report.unreadActionCount > 0 : null,
-        hasDraftComment: report && report.reportID !== activeReportID && hasComment(comments, report.reportID),
+        hasDraftComment: report && report.reportID !== activeReportID && hasComment(report.reportID),
         keyForList: report ? String(report.reportID) : personalDetail.login,
+        searchText: getSearchText(report, personalDetailList),
+        isPinned: report && report.isPinned,
     };
 }
 
+/**
+ * Rebuild the options. We are throttling this so the options are only rebuilt at most
+ * once per second. It can be expensive to rebuild options so we slow this down a bit.
+ */
+const rebuildOptions = _.throttle(() => {
+    const reportMapForLogins = {};
+    const orderedReports = lodashOrderBy(reports, ['lastVisitedTimestamp'], ['desc']);
+    allReportOptions = _.map(orderedReports, (report) => {
+        const logins = getParticipantLogins(report);
+        const hasMultipleParticipants = logins.length > 1;
+        const reportPersonalDetails = getPersonalDetailsForLogins(logins);
+        if (!hasMultipleParticipants) {
+            reportMapForLogins[logins[0]] = report;
+        }
+        return createOption(reportPersonalDetails, report);
+    });
+    allPersonalDetailsOptions = _.map(personalDetails, personalDetail => (
+        createOption([personalDetail], reportMapForLogins[personalDetail.login])
```

Comment:
So i'm wondering if we can simplify this. Only because i think giving two arguments to createOptions can get confusing. What if over here we check if `reportMapForLogins[personalDetail.login]` exists, if it does then we know that the singleUserDM is handled via reports are we can return early here. If `reportMapForLogins[personalDetail.login]` does not exist then we can createOption with just the personalDetail as an object.

Then we'd have to change the createOption in allReportOptions to just pass report and the method createOption will accept reportOrPersonalDetail object and can build its params based on what is available. Would that work?

Reply:
>if it does then we know that the singleUserDM is handled via reports are we can return early here

I might not be understanding the suggestion. But I don't think this statement is true based on how things are written now. We can't return early since we still want to build all options for both reports and personal details. Some things like the `New Chat` view don't look at reports at all.